engine/CMakeLists.txt

```diff
@@ -116,6 +116,9 @@ set(CUBOS_ENGINE_SOURCE
 	"src/input/action.cpp"
 	"src/input/axis.cpp"
 
+    "src/data/transparent_hash.cpp"
+    "src/data/transparent_equal.cpp"
```

Comment:
I think this should belong to the core library, as it is just a utility. `core::data` is a good fit!
Also, their name should probably be something like `StringTransparentHash` and `StringTransparentEqual` as they're not really 'general', they only apply to strings